sa/sa.go

```diff
@@ -135,7 +135,7 @@ func existingRegistration(tx *gorp.Transaction, id int64) bool {
 	return count > 0
 }
 
-func updateChallenges(authID string, challenges []core.Challenge, tx *gorp.Transaction) error {
+func updateChallenges(authID string, challenges []core.Challenge, tx selectExecer) error {
```

Comment:
While you're touching this function signature anyway can you make the `tx` the first arg to match the other functions?